crates/iceberg/src/spec/manifest.rs

```diff
@@ -0,0 +1,1821 @@
+        assert_eq!(manifest.metadata.schema, {
+            let fields = vec![
+                // id v_int v_long v_float v_double v_varchar v_bool v_date v_timestamp v_decimal v_ts_ntz
+                Arc::new(NestedField::optional(
+                    1,
+                    "id",
+                    Type::Primitive(PrimitiveType::Long),
+                )),
+                Arc::new(NestedField::optional(
+                    2,
+                    "v_int",
+                    Type::Primitive(PrimitiveType::Int),
+                )),
+                Arc::new(NestedField::optional(
+                    3,
+                    "v_long",
+                    Type::Primitive(PrimitiveType::Long),
+                )),
+                Arc::new(NestedField::optional(
+                    4,
+                    "v_float",
+                    Type::Primitive(PrimitiveType::Float),
+                )),
+                Arc::new(NestedField::optional(
+                    5,
+                    "v_double",
+                    Type::Primitive(PrimitiveType::Double),
+                )),
+                Arc::new(NestedField::optional(
+                    6,
+                    "v_varchar",
+                    Type::Primitive(PrimitiveType::String),
+                )),
+                Arc::new(NestedField::optional(
+                    7,
+                    "v_bool",
+                    Type::Primitive(PrimitiveType::Boolean),
+                )),
+                Arc::new(NestedField::optional(
+                    8,
+                    "v_date",
+                    Type::Primitive(PrimitiveType::Date),
+                )),
+                Arc::new(NestedField::optional(
+                    9,
+                    "v_timestamp",
+                    Type::Primitive(PrimitiveType::Timestamptz),
+                )),
+                Arc::new(NestedField::optional(
+                    10,
+                    "v_decimal",
+                    Type::Primitive(PrimitiveType::Decimal {
+                        precision: 36,
+                        scale: 10,
+                    }),
+                )),
+                Arc::new(NestedField::optional(
+                    11,
+                    "v_ts_ntz",
+                    Type::Primitive(PrimitiveType::Timestamp),
+                )),
+            ];
+            Schema::builder().with_fields(fields).build().unwrap()
+        });
+        assert!(manifest.metadata.partition_spec.fields.is_empty());
+        assert!(manifest.metadata.content == ManifestContentType::Data);
+        assert!(manifest.metadata.format_version == FormatVersion::V2);
+        // test entries
+        assert!(manifest.entries.len() == 1);
+        let entry = &manifest.entries[0];
+        assert!(entry.status == ManifestStatus::Added);
+        assert!(entry.snapshot_id == Some(0));
+        assert!(entry.sequence_number == Some(1));
+        assert!(entry.file_sequence_number == Some(1));
```

Comment:
This is some interesting test-data 🤔 The status is `added`, so this means a new file which typically doesn't set the snapshot-id, sequence-number, and file-sequence-number since they should be inherited. This way we don't have to rewrite the manifest after a failed conflict (because a new snapshot-id and sequence-number is assigned).